keystoremanager/keystoremanager.go

```diff
@@ -205,27 +209,26 @@ func (km *KeyStoreManager) ValidateRoot(root *data.Signed, dnsName string) error
 
 		// Check to see if there is an exact match of this certificate.
 		// Checking the CommonName is not required since ID is calculated over
-		// Cert.Raw. It's included to prevent breaking logic with changes of how the
-		// ID gets computed.
-		_, err = km.certificateStore.GetCertificateByKeyID(leafID)
+		// Cert.Raw. It's included to prevent breaking logic with changes of
+		// how the ID gets computed.
+		_, err = km.certificateStore.GetCertificateByCertID(leafID)
 		if err == nil && leafCert.Subject.CommonName == dnsName {
-			certs[keyID] = rootSigned.Keys[keyID]
+			validKeys[keyID] = rootSigned.Signed.Keys[keyID]
 		}
 
-		// Check to see if this leafCertificate has a chain to one of the Root CAs
-		// of our CA Store.
+		// Check to see if this leafCertificate has a chain to one of the Root // CAs of our CA Store.
```

Comment:
Is this supposed to support building a chain that contains an intermediate we've never seen? Seems to require the intermediate to already be in the caStore?